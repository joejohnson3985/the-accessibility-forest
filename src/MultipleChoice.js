import React, { Component } from 'react';
import MultipleChoiceItem from './MultipleChoiceItem.js';
import './MultipleChoice.scss';

class MultipleChoice extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showPopUp: false
    }
  }

  componentDidMount() {
    this.displayAnswers();
  }

  displayAnswers() {
    const answersDisplay = [];
    let answerOptions = this.props.wrongAnswers;
    answerOptions.push(this.props.currentAnswer)
    answerOptions.sort()
    for(let i = 0; i < 4; i++) {
      answersDisplay.push(<MultipleChoiceItem choice={answerOptions[i]} key={i} handleAnswer={this.handleAnswer}/>)
    }
    return answersDisplay;
  }

  handleAnswer = (answer) => {
    this.props.displayNextTerm(answer);
  }

  checkAnswer = (answer) => {
    if(answer === this.props.currentAnswer) {
      this.props.displayNextTerm(answer);
    } else {
      this.props.getCurrentTerm();
    }
  }

  render() {
    return (
      <div className='multiple-choice-container'>
        {this.displayAnswers()}
      </div>
    )
  }
}

export default MultipleChoice;
