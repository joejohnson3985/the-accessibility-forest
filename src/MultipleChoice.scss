@import './_variables.scss';

.multiple-choice-container {
  margin: 5px;
}