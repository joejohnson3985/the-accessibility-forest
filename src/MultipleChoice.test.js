import React from 'react';
import MultipleChoice from './MultipleChoice';
import { shallow } from 'enzyme';

const currTerm = '<address>';
const wrongAnswers = ['<article>', '<code>', '<nav>'];
const mockFunc = jest.fn();

describe('MultipleChoice', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
        <MultipleChoice 
          currentAnswer={currTerm} 
          wrongAnswers={wrongAnswers}
          displayNextTerm={mockFunc}
        />
      )
  });

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Should call the function displayNextTerm inside handleAnswer', () => {
    wrapper.instance().handleAnswer();
    expect(mockFunc).toBeCalled();
  })
})