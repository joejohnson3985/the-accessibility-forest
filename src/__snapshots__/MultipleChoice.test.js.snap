// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultipleChoice Should match the snapshot 1`] = `
<div
  className="multiple-choice-container"
>
  <MultipleChoiceItem
    choice="<address>"
    handleAnswer={[Function]}
    key="0"
  />
  <MultipleChoiceItem
    choice="<article>"
    handleAnswer={[Function]}
    key="1"
  />
  <MultipleChoiceItem
    choice="<code>"
    handleAnswer={[Function]}
    key="2"
  />
  <MultipleChoiceItem
    choice="<nav>"
    handleAnswer={[Function]}
    key="3"
  />
</div>
`;
